sycl/doc/design/SYCLBINDesign.md

```diff
@@ -0,0 +1,280 @@
+| `uint64_t` | Byte size of the raw IR bytes in the binary byte table.                  |
+
+
+##### IR module metadata
+
+An IR module metadata entry contains a single property set with the identifying
+name "SYCLBIN/ir module metadata", as described in the
+[PropertySets.md](PropertySets.md#syclbinir-module-metadata) design document.
+
+
+#### Native device code image
+
+An native device code image contains the binary data for the corresponding
+module AOT compiled for a specific device, identified by the architecture
+string.
+
+
+##### Native device code image header
+
+A native device code image header contains the following fields in the stated
+order:
+
+| Type       | Description                                                                         |
+| ---------- | ----------------------------------------------------------------------------------- |
+| `uint64_t` | Byte offset of the metadata from the start of the metadata byte table.              |
+| `uint64_t` | Byte size of the metadata in the metadata byte table.                               |
+| `uint64_t` | Byte offset of the device code image bytes from the start of the binary byte table. |
+| `uint64_t` | Byte size of the device code image bytes in the binary byte table.                  |
+
+
+##### Native device code image metadata
+
+A native device code image metadata entry contains a single property set with
+the identifying name "SYCLBIN/native device code image module metadata", as
+described in the
+[PropertySets.md](PropertySets.md#syclbinnative-device-code-image-metadata)
+design document.
+
+
+### Byte tables
+
+A byte table contains dynamic data, such as metadata and binary blobs. The
+contents of it is generally referenced by an offset specified in the headers.
+
+
+### SYCLBIN version changelog
+
+The SYCLBIN format is subject to change, but any such changes must come with an
+increment to the version number in the header and a subsection to this section
+describing the change.
+
+Additionally, any changes to the property set structure that affects the way the
+runtime has to parse the contained property sets will require an increase in the
+SYCLBIN version. Adding new property set names or new predefined properties only
+require a SYCLBIN version change if the the SYCLBIN consumer cannot safely
+ignore the property.
+
+
+#### Version 1
+
+ * Initial version of the layout.
+
+
+## Clang driver changes
+
+The clang driver needs to accept the following new flags:
+
+<table>
+<tr>
+<th>Option</th>
+<th>Description</th>
+</tr>
+<tr>
+<td>`-fsyclbin`</td>
+<td>
+If this option is set, the output of the invocation is a SYCLBIN file with the
+.syclbin file extension. This skips the host-compilation invocation of the typical
+`-fsycl` pipeline, instead passing the output of the clang-offload-packager
+invocation to clang-linker-wrapper together with the new `--syclbin` flag.
+
+Setting this option will imply `-fsycl` and override `-fsycl-device-only`.
```

Comment:
Could you please clarify what do you mean by "override `-fsycl-device-only`"? Isn't skipping host compilation _implies_ `-fsycl-device-only`?

Reply:
I suppose so. To some extend I think the terminology overlaps a little. They are not really mutually exclusive, it is just "`-fsycl-device-only` with more". I'll change it to just "implies". @mdtoguchi - Is there a precedence for the terminology used here?

Reply:
I think the intention of the wording here is that use of `-fsyclbin` will only produce the SYCLBIN file.  Expectation of using `-fsycl-device-only` is produce just the device binary.  If they are used together, `-fsycl-device-only` is overridden from a driver function perspective.